build-analyze.ps1

```diff
@@ -0,0 +1,37 @@
+$ErrorActionPreference = 'Stop'
+
+# SonarCloud needs a full clone to work correctly but some CIs perform shallow clones
+# so we first need to make sure that the source repository is complete
+git fetch --unshallow
+
+$SONAR_SERVER_URL = "https://sonarcloud.io"
+#$SONAR_TOKEN = # Access token coming from SonarCloud projet creation page. In this example, it is defined in the environement through a Github secret.
+$SONAR_SCANNER_VERSION = "4.6.1.2450" # Find the latest version in the "Windows" link on this page:
+                                      # https://docs.sonarcloud.io/advanced-setup/ci-based-analysis/sonarscanner-cli/
+$BUILD_WRAPPER_OUT_DIR = "build_wrapper_output_directory" # Directory where build-wrapper output will be placed
+
+mkdir $HOME/.sonar
+
+# Download build-wrapper
+$path = "$HOME/.sonar/build-wrapper-win-x86.zip"
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+(New-Object System.Net.WebClient).DownloadFile("$SONAR_SERVER_URL/static/cpp/build-wrapper-win-x86.zip", $path)
+Add-Type -AssemblyName System.IO.Compression.FileSystem
```

Comment:
```suggestion

# Prepare downloading and extraction of build-wrapper and sonar-scanner
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
Add-Type -AssemblyName System.IO.Compression.FileSystem

# Download build-wrapper
$path = "$HOME/.sonar/build-wrapper-win-x86.zip"
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
(New-Object System.Net.WebClient).DownloadFile("$SONAR_SERVER_URL/static/cpp/build-wrapper-win-x86.zip", $path)
```

Reply:
Ok
(I've modified your suggestion because the line with `Net.ServicePointManager` appeared two times)
```